lightning/src/util/test_utils.rs

```diff
@@ -341,15 +341,13 @@ fn get_dummy_channel_update(short_chan_id: u64) -> msgs::ChannelUpdate {
 pub struct TestRoutingMessageHandler {
 	pub chan_upds_recvd: AtomicUsize,
 	pub chan_anns_recvd: AtomicUsize,
-	pub request_full_sync: AtomicBool,
```

Comment:
Well, the filter is currently already sent in `peer_connected()` and since `request_full_sync` was not used at all, I figured it would sense to simply remove it rather than to add much more logic to the so-far pretty slim `TestRoutingMessageHandler`. Or are you suggesting to introduce `request_full_sync` to other places, e.g., `NetGraphMsgHandler`, so it could be used by `should_request_full_sync()`?

Reply:
Wait, who is sending the filter in `peer_connected` when we're using a `TestRoutingMessageHandler`? Its `peer_connected` method is empty.

Reply:
Ah, now I get what you're saying. You're right, its probably easiest here to let the `TestRoutingMessageHandler` implementation of `peer_connected` mirror the functionality of `NetGraphMsgHandler`.

Reply:
I went that route (replicating the `peer_connected` functionality in `TestRoutingMessgaeHandler`) and it generally seems to work nicely. However, now the second peer receives only 86/100 channel updates and 43/50 channel announcements, i.e., the test now fails at the last two `assert_eq`s.  I currently struggle to find why this would be the case, since the logs show that the gossip is indeed just postponed until the peers have received `gossip_timestamp_filter` messages. I'll come around to continue staring at the code next week.

Reply:
Yes, and before I adapt the test I at least would like to understand why suddenly not the same number of channel updates are delivered, because its not intuitive to me from the changes I made...

Reply:
In https://github.com/lightningdevkit/rust-lightning/pull/1452/commits/7a99de2ae0e67cd64c754e52ad87d760fb8dac66 I now moved some things in the loop around. This makes the test succeed hopefully without reducing the coverage too much (i.e., it now checks for B's empty `outbound_data` once in the end, not in every iteration).